pgmpy/inference/Sampling.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+
+import networkx as nx
+from pandas import DataFrame
+from pgmpy.inference import Inference
+from pgmpy.models import BayesianModel
+from pgmpy.utils.mathext import sample_discrete
+
+
+class BayesianModelSampling(Inference):
+    """
+    Class for sampling methods specific to Bayesian Models
+
+    Parameters
+    ----------
+    model: instance of BayesianModel
+        model on which inference queries will be computed
+
+
+    Public Methods
+    --------------
+    forward_sample(size)
+    """
+    def __init__(self, model):
+        if not isinstance(model, BayesianModel):
+            raise TypeError("model must an instance of BayesianModel")
+        super(BayesianModelSampling, self).__init__(model)
+        self.topological_order = nx.topological_sort(model)
+        self.cpds = {}
+        for node in model.nodes():
+            self.cpds[node] = model.get_cpds(node)
+
+    def forward_sample(self, size=1):
```

Comment:
@pratyakshs One another way to computes samples would be by looping over the variables rather than looping over the number of samples we want. Taking the example in the doctest and let's say we want 1000 samples, we will start with 'diff' and compute all the 1000 samples for it by a single `sample_discrete` call, then can compute all the 1000 samples for 'intel` and since now we have the states of all the 1000 samples of `intel` and `diff` we can compute the values for `grade`. 
This method should be much faster and will also give us some space to vectorize computations. For starters for a call on 5 vars for 10000 samples we will need to call `sample_discrete` only 5 times instead of 50000 times.
Also, this method will give us space for memoization reduce calls.